hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/SchemaTransaction.java

```diff
@@ -361,6 +377,47 @@ private BackendEntry serialize(SchemaElement schema) {
         }
     }
 
+    public Id validOrGenerateId(HugeType type, Id id, String name) {
+        if (id != null) {
+            this.checkIdAndUpdateNextId(type, id, name);
+        } else {
+            if (Graph.Hidden.isHidden(name)) {
```

Comment:
boolean forSystem = Graph.Hidden.isHidden(name)
and pass into checkIdAndUpdateNextId()